site/content/en/latest/design/pprof.md

```diff
@@ -1,15 +1,19 @@
 ---
```

Comment:
prefer to move this out of this PR

Reply:
The debug server changes are related to this PR, we need to separate the debug server (including pprof) out of admin server.

Reply:
why?

Reply:
debug server for now is mostly used for pprof, may be exposed more debugging features in the future, it is optional and not recommended to be used in production env, but the metrics should always be enabled.

It is better to use a sparate port for debugging/pprof with admin port.

And we can remove the logics here: 

<img width="854" alt="image" src="https://github.com/envoyproxy/gateway/assets/48784001/4ef6bcab-5caf-45cf-862c-d8dd5f4cd17c">

By using controller-runtime metrics server, the reuse the logics in controller-runtime to enable pprof, and it is also not allowed in controller-runtime to expose metrics and pprof in the same port.

Reply:
One thing per PR, let focus on metrics in this PR.
You can do pprof in a seperated one.

Reply:
Okay, make sense

Reply:
https://github.com/envoyproxy/gateway/pull/1967